src/server/game/AI/CreatureAI.cpp

```diff
@@ -170,10 +207,50 @@ void CreatureAI::EnterEvadeMode()
         }
     }
 
-    Reset();
-
-    if (me->IsVehicle()) // use the same sequence of addtoworld, aireset may remove all summons!
-        me->GetVehicleKit()->Reset(true);
+    // despawn bosses at reset - only verified tbc/woltk bosses with this reset type - add bosses in last line respectively (dungeon/raid) and increase array limit
+    static constexpr std::array<uint32, 24> bosses = {
```

Comment:
Could you add a `//@todo: Turn into a flags_extra in creature_template` so that we in the future can remove the hardcoded stuff?

Reply:
Will do